src/SensorUtils.cpp

```
@@ -68,6 +70,72 @@ double PhaseAngle(const std::vector<double> &observerBodyFixedPosition,
 
 
 
+
+/**
+ * @brief rectangular2latitudinal
+ * @author Tyler Wilson
+ * @param rectangularCoords
+ * @return  Given a set of J2000 coordinates, returns [Range,RightAscension,Declination] in Radians
+ */
+
+
+vector<double> rectangular2latitudinal(const vector<double> rectangularCoords){
+
+  vector<double> radiusLongLat{0.0,0.0,0.0};
+
+
+  double d1 = abs(rectangularCoords[0]);
+  double d2 = abs(rectangularCoords[1]);
+
+  d1 = max(d1,d2);
+  d2 = abs(rectangularCoords[2]);
+
+  double big = max(d1,d2);
+
+  if (big > 0.0) {
+    double x = rectangularCoords[0]/big;
+    double y = rectangularCoords[1]/big;
+    double z = rectangularCoords[2]/big;
+
+    //Radius
+    radiusLongLat[0] = big*sqrt(pow(x,2.0)+pow(y,2.0)+pow(z,2.0));
+
+    //Latitude
+    radiusLongLat[2] = atan2(z,sqrt(pow(x,2)+pow(y,2)));
+
+    //Longitude
+    if (abs(rectangularCoords[0]) <= DBL_MIN && abs(rectangularCoords[1]) <=DBL_MIN) {
+      radiusLongLat[1] = 0.0;
+      }
+    else {
+      radiusLongLat[1] = atan2(rectangularCoords[1],rectangularCoords[0]);
+    }
+  }
+
+   return radiusLongLat;
+
+ }
```

This whole method seems really complex for what it's doing. Disregarding exceptional cases such as when (0,0,0) is passed in, this method is two normalizations and two inverse trig functions. What are the cases in which we want to return the error value of (0,0,0)? Splitting those cases into explicit checks would help make this more readable. We also need to document those cases in the function documentation.